### PR TITLE
🐛Source Amazon Seller Partner: Implement slicing for report streams

### LINKED ISSUE #14734: Amazon Seller Partner (Cloud Only) Connection Error
**Please note, this connector is in Alpha. Read more here [on expectations](https://docs.airbyte.com/integrations/).**

We're aware that there is a connection error impacting all Amazon Seller Partner source connectors in Airbyte Cloud.

The team is actively working on resolving this with the Amazon team directly but are facing some challenges with the approval process. Please feel free to follow this issue to be notified of any updates.

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/alpha-beta-issues/issues/257

- Implement stream slicing for report streams since it looks like Amazon API aborts reports requesting a period of more than 30 days
- Updated SAT to high strictness 
- Deleted `GET_FBA_INVENTORY_AGED_DATA` and `GET_VENDOR_INVENTORY_HEALTH_AND_PLANNING_REPORT` as deprecated - https://github.com/amzn/selling-partner-api-docs/discussions/2165

### CODE REVIEW

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py
Comment (bazarnov) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py:68]: Can we make the description more human-readable for Customers?

**Review by bazarnov — APPROVED** [review]

**Comment by animer3009** [comment]
Are you sure that it can't be more than 30 days?

Some reports can requested with 60 day period.

For example you can find `You can request up to 60 days of data in a single report` in descriptions here:

https://developer-docs.amazon.com/sp-api/docs/report-type-values

So please do not restrict period_in_days.

**Comment by davydov-d** [comment]
> @davydov-d @bazarnov
> 
> Are you sure that it can't be more than 30 days?
> 
> Some reports can requested with 60 day period.
> 
> For example you can find `You can request up to 60 days of data in a single report` in descriptions here:
> 
> https://developer-docs.amazon.com/sp-api/docs/report-type-values
> 
> So please do not restrict period_in_days.

@animer3009 well I can't be sure of anything since this is Amazon API :sweat_smile: 
thanks for your feedback, I guess we'll need to do a more intense research

**Comment by animer3009** [comment]
@lazebnyi @bazarnov
Guys are you anyway trying to push changes to restrict period_in_days with 30?
Have you seen my comment?
https://github.com/airbytehq/airbyte/pull/16430#issuecomment-1303013493

**Comment by lazebnyi** [comment]
> @lazebnyi @bazarnov
> 
> Guys are you anyway trying to push changes to restrict period_in_days with 30?
> 
> Have you seen my comment?
> 
> https://github.com/airbytehq/airbyte/pull/16430#issuecomment-1303013493

Yes, we saw your comments. Thanks for that🙏 We will try implement different rages for different streams. That's not final version of PR. Just draft 😊

File: airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml
Comment (sherifnada) [inline airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml:33]: why are there no records? Can you give some context that helps the next person understand why we can't have records here?

**Review by sherifnada — APPROVED** [review]

**Review by girarda — APPROVED** [review]

File: airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml
Comment (alafanechere) [inline airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml:2]: I don't think it is fair to enable a high test strictness level if so many streams are declared empty in the basic read. @YowanR could you evaluate if we could make seeding efforts to have more test data and less "empty streams"?

**Review by alafanechere — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml
Comment (alafanechere) [reply airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml:2]: @davydov-d I would suggest working on the `high` test strictness level from another PR so that your other changes are not blocked by my request.

Comment (davydov-d) [reply airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml:2]: I'm not working on this PR, @lazebnyi is :)

Comment (YowanR) [reply airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml:2]: This is also not a GA connector. Agreed with @alafanechere. Let's not enable high strictness test here and do it in another PR. @lazebnyi please create another issue for this and we'll have the Unblockers look at adding more data

File: airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-amazon-seller-partner/acceptance-test-config.yml:33]: Added TODO comment

**Review by alafanechere — APPROVED** [review]
The `acceptance-test-config.yml`file looks good to me now that the `test_strictness_level` is not set to `high`. 
You are introducing a small backward incompatible change:
Users with configuration with the `period_in_days` to a value > 30 will likely get a check command failure.
Please sync up with @airbytehq/cloud-support before merging to make sure outreach is sent to the customer using this connector.

**Comment by TBernstein4** [comment]
@alafanechere I just checked and there is only one workspace on Cloud that has this connector setup and it is internal (@amaliaroye) so there is no outreach necessary from TCS.

**Review by tybernstein — APPROVED** [review]
No outreach necessary as there are no Cloud users impacted by this update.

**Comment by animer3009** [comment]
Guys!!!
You are still going to merge changes which restrict `period_in_days` with 30 day.
I will repeat the same.
1. Some reports allow 30 day +
2. We are using Airbyte ASP for the production. ASAP you will merge this, we will face issues! As sometimes we are requesting 31 and 31 day + reports.

So the main question is:
Why you are not keeping possibility to request 30 day + reports. Just add a notice that recommended use 30! Or restrict it in cloud side...

**Comment by lazebnyi** [comment]
@animer3009 Thanks! We decide increase max value for period_in_days to 90 days. But for `GET_AMAZON_FULFILLED_SHIPMENTS_DATA_GENERAL` and `GET_FLAT_FILE_RETURNS_DATA_BY_RETURN_DATE` streams maximum value for `period_in_days` 30 days and 60 days. 
So, for any value that exceeds the limit, the `period_in_days` will be automatically reduced to the limit for the stream.

**Pull request merged**
